src/routes/api/playlists/library/server.test.ts

```diff
@@ -0,0 +1,94 @@
+import { expect, test } from 'vitest';
+import { GET, OPTIONS } from './+server';
+import type { RequestEvent } from './$types';
+
+test('Options method contains correct headers', () => {
+  const response = OPTIONS();
+  expect(response.headers.get('Access-Control-Allow-Methods')).toBe('GET');
+});
+
+test('GET method with no filters returns all library playlist data', async () => {
+  const urlstring = 'https://test.com/api/playlists/library';
+  const mockRequestEvent = {
+    request: new Request(urlstring),
+    url: new URL(urlstring),
+    params: {},
+  } as unknown as RequestEvent;
```

Comment:
had to add this weird cast bc we needed the params in the `RequestEvent` but not everything else so

Reply:
If you did the casting when you call the function, would that work? Seems simpler on the test I did:
`const response = await GET({ request } as RequestEvent);`

Reply:
i saw that's how it was done in your other tests, but then it wouldn't let me set those params... i will try one more time!